src/main/kotlin/org/rust/ide/actions/RsCreateCrateAction.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import org.rust.cargo.icons.CargoIcons
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.settings.toolchain
+import org.rust.cargo.runconfig.command.RunCargoCommandActionBase
+import org.rust.cargo.toolchain.RustToolchain
+import org.rust.ide.actions.ui.CargoNewCrateDialog
+import org.rust.openapiext.pathAsPath
+
+
+class RsCreateCrateAction : RunCargoCommandActionBase(CargoIcons.ICON) {
+    override fun actionPerformed(e: AnActionEvent) {
+        val dataContext = e.dataContext
+        val project = CommonDataKeys.PROJECT.getData(dataContext) ?: return
+        val root = CommonDataKeys.VIRTUAL_FILE.getData(dataContext) ?: return
```

Comment:
Since the parent already enables the action only if there is a Cargo project and I handle running the action on both files and directories, it should be available always, or not?